crates/chain/src/sparse_chain.rs

```diff
@@ -456,7 +457,27 @@ impl<P: core::fmt::Debug> core::fmt::Display for UpdateError<P> {
 #[cfg(feature = "std")]
 impl<P: core::fmt::Debug> std::error::Error for UpdateError<P> {}
 
-impl<P: ChainPosition> SparseChain<P> {
+impl<P> ChainOracle for SparseChain<P> {
```

Comment:
Concept question. Now that we have `LocalChain`, do we also intend to use `SparseChain` too? If so, whats the difference between the two?

Reply:
We need to think about whether `SparseChain` is still useful. The difference between the two is that `LocalChain` does not track txids in chain, and only blocks.

Reply:
Removed.